### PR TITLE
Issue #1952: Change sort settings on domain requests admin (rjm)

### LINKED ISSUE #1952: Admin: make `submission date` the default sort for domain requests list page
### Problem:
In the domain requests list page, currently, this `Submission Date` column is far to the right and requires horizontal scrolling to see it. In order to help the analyst prioritize incoming requests, they would prefer to see the Submission Date more easily. 

Also, analysts would prefer that the most recent submissions appear at the top of this list, so they may easily see which tickets have not been triaged and assigned to an analyst for review. This means that the oldest requests would be at the bottom of the list.

### Solution:
We will move the submission date column right next to the domain name column in the view. The default view for this page will then be sorted in reverse chronological order by submission date.

The second level sort will be by domain name, in ascending alphabetical order.

### Acceptance criteria

- [ ] On the /admin/registrar/domainrequest/ page, the default view is sorted by `Submission Date`, in reverse chronological order.
- [ ] In that same view, the `Submission Date` column appears in the second position, preceded by `Requested Domain` and followed by `Status` columns.

### Additional context

Request came from the 3/27/24 Product Huddle, [with Bookmark to notes here](https://docs.google.com/document/d/1jatolB9YaQLITwqx25y0B1u5J23S_J19PawXtIaov8o/edit#bookmark=id.3lxagvi8bl8u).

This issue has come up several times, and should be considered a priority.

### Links to other issues

Relates to: #1838. This ticket has been inspired by that work.

### PR DESCRIPTION
**Pull request by rachidatecs**
## Ticket

Resolves #1952 

## Changes

- Sort primarily by reverse submission date, then alphabetically by domain name

#### Satisfied acceptance criteria and met development standards

- [x] Met the acceptance criteria, or will meet them in a subsequent PR
- [x] Created/modified automated tests
- [x] Added at least 2 developers as PR reviewers (only 1 will need to approve)
- [x] Messaged on Slack or in standup to notify the team that a PR is ready for review
- [ ] Changes to “how we do things” are documented in READMEs and or onboarding guide
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the associated migrations file has been commited.

#### Ensured code standards are met (Original Developer)

- [x] All new functions and methods are commented using plain language
- [ ] Did dependency updates in Pipfile also get changed in requirements.txt?
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values

#### Validated user-facing changes (if applicable)

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)
- [ ] Add at least 1 designer as PR reviewer

#### Reviewed, tested, and left feedback about the changes

- [ ] Pulled this branch locally and tested it
- [ ] Reviewed this code and left comments
- [ ] Checked that all code is adequately covered by tests
- [ ] Made it clear which comments need to be addressed before this work is merged
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the associated migrations file has been commited.

#### Ensured code standards are met (Code reviewer)

- [ ] All new functions and methods are commented using plain language
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values
- [ ] (Rarely needed) Did dependency updates in Pipfile also get changed in requirements.txt?

#### Validated user-facing changes as a developer

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Meets all designs and user flows provided by design/product
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers, the suggestion is to use ones that the developer didn't (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

**Note:** Multiple code reviewers can share the checklists above, a second reviewers should not make a duplicate checklist

#### Verified that the changes match the design intention

- [ ] Checked that the design translated visually
- [ ] Checked behavior
- [ ] Checked different states (empty, one, some, error)
- [ ] Checked for landmarks, page heading structure, and links
- [ ] Tried to break the intended flow

#### Validated user-facing changes as a designer

- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

### CODE REVIEW

**Review by kristinacyin — APPROVED** [review]
LGTM! 👍 It's really funny that requests go back to 70s haha

File: src/registrar/admin.py
Comment (rachidatecs) [reply src/registrar/admin.py:1195]: Where are the domain requests select2 fields?

Comment (rachidatecs) [reply src/registrar/admin.py:1195]: Does not look like we're using those on domain requests. If that's the case, I think we should make a note in the code somewhere about your guess and leave it alone for now

Comment (rachidatecs) [reply src/registrar/admin.py:1195]: That's a good catch! However, analysts have only read-only access, and I can't think of a scenario where a superadmin has to change the domain request associated with a domain information. In my opinion, any fix here is not worth the overhead we'll introduce. That combo box is also searchable after all, in case there's ever an instance where a superadmin actually needs to do this.

Defer to @abroddrick

**Review by therealslimhsiehdy — APPROVED** [review]
Code and tests LGTM -- I'll give an approval and defer to Alysia (notified her of the discussion y'all have) for the final approval decision or if changes are needed.

File: src/registrar/admin.py
Comment (abroddrick) [reply src/registrar/admin.py:1195]: I completely agree rachid! Not worth the time to update this, nor should we ever change the application assigned to a domain and when in doubt I would NEVER actually even scroll in that drop down so it doesn't matter how it's sorted.

**Pull request merged**
